statsmodels/regression/linear_model.py

```diff
@@ -1511,7 +1512,7 @@ def __init__(self, model, params, normalized_cov_params=None, scale=1.,
             setattr(self, key, kwargs[key])
 
     def __str__(self):
-        self.summary()
+        return self.summary()
```

Comment:
actually, I think this is wrong
summary() returns a simple table and not a string
I guess something like `as_text()`  needs to be added

I guess the unit test that you added doesn't catch this because of the Summary instance has it's own `__str__` 

actually I don't understand why this works